modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/upgrade/v3_1_0/UrlSubjectUpgradeProcess.java

```diff
@@ -89,22 +63,33 @@ private String _getUrlSubject(long id, String subject) {
 
 	private void _populateUrlSubject() throws Exception {
 		try (PreparedStatement ps1 = connection.prepareStatement(
-				"select messageId, subject from MBMessage where (urlSubject " +
-					"is null) or (urlSubject = '')");
+				"select messageId, subject from MBMessage order by subject, " +
+					"messageId asc");
 			ResultSet rs = ps1.executeQuery();
 			PreparedStatement ps2 = AutoBatchPreparedStatementUtil.autoBatch(
 				connection.prepareStatement(
 					"update MBMessage set urlSubject = ? where messageId = " +
 						"?"))) {
 
+			int count = 0;
+			String currUrlSubject;
+			String prevUrlSubject = null;
+
 			while (rs.next()) {
 				long messageId = rs.getLong(1);
 				String subject = rs.getString(2);
 
-				String uniqueUrlSubject = _findUniqueUrlSubject(
-					connection, _getUrlSubject(messageId, subject));
+				currUrlSubject = _getUrlSubject(messageId, subject);
 
-				ps2.setString(1, uniqueUrlSubject);
+				if (StringUtil.equals(prevUrlSubject, currUrlSubject)) {
+					ps2.setString(
+						1, currUrlSubject + StringPool.DASH + ++count);
```

Comment:
Avoid using pre/postincrement as part of bigger expression. Put the increment in its own statement.

Reply:
Maybe, change the whole thing to something like this:
```
if (StringUtil.equals(previousURLSubject, curURLSubject)) {
    count++;
    suffix = StringPool.DASH + count;
}
else {
    count = 0;
    suffix = StringPool.BLANK;
}

ps2.setString(1, curURLSubject + suffix);
```